### PR TITLE
Canvas sso and launch styling

### PR DESCRIPTION
**Pull request by brendanshean**
## WHAT
Get the CSS styling on the Canvas SSO consistent with the `/sessions/new`

## WHY
The UX should be consistent especially with authentication/authorization pages.

## HOW
1. Extract out `on_sign_up_or_log_in?` helper to determine which header & footer should be used.
1. Add a footer/base partial that contains minimal footer elements
1. Add inline styling for the launch.html.erb (inline styling is required here since canvas restricts <script> tags)  Screenshot 2023-06-12 at 12 03 25 PM

1. Render footer/base with the sso view. Screenshot 2023-06-12 at 9 19 45 AM
Screenshot 2023-06-12 at 9 19 51 AM

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  No.  Just view changes.  Manual check
Have you deployed to Staging? | Not yet - deploying now!
Self-Review: Have you done an initial self-review of the code below on Github? | YES
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | YES

### CODE REVIEW

**Review by dandrabik — APPROVED** [review]
:+1: Looks good to me, thanks for putting this together. My only notes:

1. We need to do something similar (but even more simple) to the XML page in Canvas (we might need to inline styles there if we can't pull external css).
2. Before launch we should do some testing to see if page load time is a concern and we should strip the page down more, e.g. not loading full js/css files or adjusting when the button click /redirect is triggered, etc.

Thanks for pushing this along!

**Comment by brendanshean** [comment]
@dandrabik Re-requesting for added launch styling

File: services/QuillLMS/app/views/canvas_integration/lti/launch.html.erb [hunk 1]
```diff
@@ -1 +1,59 @@
...
+      padding: 10px 20px;
+      text-align: center;
+      text-decoration: none;
+      background-color: #f7f5f5;
+      border: solid 1px #e0e0e0;
+      border-radius: 5px;
+    }
+  </style>
+
+</style>t
```
Comment (dandrabik) [inline services/QuillLMS/app/views/canvas_integration/lti/launch.html.erb]: extra </style> typo?

File: services/QuillLMS/app/views/canvas_integration/lti/launch.html.erb [hunk 2]
```diff
@@ -1 +1,59 @@
...
+  .logo-only {
+    width: 100%;
+    display: flex;
+    justify-content: center;
+    align-items: center;
+  }
+
+  .text-center { text-align: center }
+
+  .account-form {
```
Comment (dandrabik) [inline services/QuillLMS/app/views/canvas_integration/lti/launch.html.erb:13]: Let's unnest the CSS for better backwards compatibility with older browsers. https://caniuse.com/css-nesting  

e.g.
```css
.account-form {

}

.account-form .account-container {

}
...

**Review by dandrabik — APPROVED** [review]
Looks good! Small note about a typo and un-nesting the inline CSS, but otherwise 🚢

**Review by eadams17 — APPROVED** [review]

**Pull request merged**